### ISSUE TITLE
Shows unknown "Called from" stack frame in crash report

### ISSUE THREAD

**Issue by sindresorhus** [getsentry/sentry-cocoa#1620]
### Environment

SaaS (https://sentry.io/)

### Version

Sentry Cocoa 7.5.2

### Steps to Reproduce

- Go to https://sentry.io/organizations/sindresorhus/issues/2833675135/events/8f9b4afd74da49fbb3e3fef911c4433e/

### Expected Result

I expected to see the name of each stack frame.

### Actual Result

The first stack frame is "unknown" and "redacted".
<img width="944" alt="Screen Shot 2021-12-06 at 18 10 10" src="https://user-images.githubusercontent.com/170270/144890540-f58e836a-93af-460b-b728-ae6b9bf1ccd1.png">

This is a macOS app using the SwiftUI `App` protocol.

**Comment by philipphofmann** [comment]
@jan-auer the logic for loading binary images is in [SentryCrashDynamicLinker](https://github.com/getsentry/sentry-cocoa/blob/7cf0f7441aed40e4c344404d10172547fb3edd4a/Sources/SentryCrash/Recording/Tools/SentryCrashDynamicLinker.c#L336). The original file [KSDynamicLinker](https://github.com/kstenerud/KSCrash/commits/master/Source/KSCrash/Recording/Tools/KSDynamicLinker.c) doesn't contain any new patches that we don't have yet. I already applied the latest patch of `accumulative patch` with https://github.com/getsentry/sentry-cocoa/pull/1596.

**Comment by jan-auer** [comment]
Sorry, I made a mistake when looking at this the first time. `libdyld` is there in all cases, it's just unreferenced and the UI hides it. If you reset filters you'll find it. 

This means, the bottom frame returned by the stack walker is either wrong, or it points at an image that's unloaded at the time of the crash. If you look at the address, it lies shortly after the `Sentry` framework, which suggests that this is a user-space library.

**Comment by sindresorhus** [comment]
Please keep this open.

**Comment by amadeu01** [comment]
@philipphofmann do you have a plan to fix this issue ?

**Comment by philipphofmann** [comment]
No, not yet. We are going to discuss this again internally, as I also find it annoying, and we should fix it.

**Comment by amadeu01** [comment]
@philipphofmann have you had any update on that one?

**Comment by philipphofmann** [comment]
@amadeu01, sadly not yet, no.

**Comment by philipphofmann** [comment]
We could do something similar as [this code does](https://github.com/bugsnag/bugsnag-cocoa/blob/c730af7a4676617d82394782300df997d9a356cd/Bugsnag/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSMachHeaders.c#L85-L109), but we must respect the license.

**Comment by amadeu01** [comment]
@philipphofmann do you have any update?

**Comment by brustolin** [comment]
Hello @amadeu01, we have no updates for this issue.

**Comment by philipphofmann** [comment]
I still see this happening for our iOSSwift sample app
![CleanShot 2023-10-31 at 15 51 08@2x](https://github.com/getsentry/sentry-cocoa/assets/2443292/3ca81ddd-c37c-4963-ac29-25b025975e2f)


and for our macOS sample app: https://sentry-sdks.sentry.io/issues/4409194420/events/eca0ab201ffb4aa79ca342b207eebfaa/

![CleanShot 2023-10-31 at 15 48 53@2x](https://github.com/getsentry/sentry-cocoa/assets/2443292/bd352e32-c3af-4a7d-8bf6-a1ffb979fff5)

cc @kahest and @brustolin, we need look into this.

**Comment by kahest** [comment]
We'll have a look again at the root cause. If there is no pragmatic/reasonable way to solve this we may look at custom rules to not show this specific frame since it doesn't provide any value.

**Comment by philprime** [comment]
@kahest is this issue still relevant? There has not been any activity since Nov 2023

**Comment by kahest** [comment]
@philprime yes, it's still happening and we should find out if there's a reasonable-effort way to improve this
